src/AddressUkPafLookupView.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+
+namespace Gcd\UkAddressLookup;
+
+
+use Rhubarb\Leaf\Presenters\Controls\Buttons\Button;
+use Rhubarb\Leaf\Presenters\Controls\ControlView;
+use Rhubarb\Leaf\Presenters\Controls\Selection\DropDown\DropDown;
+use Rhubarb\Leaf\Presenters\Controls\Text\TextBox\TextBox;
+
+class AddressUkPafLookupView extends ControlView
+{
+    protected $htmlType = "address";
+
+    public function __construct($htmlType = "address")
+    {
+        $this->htmlType = $htmlType;
+
+        $this->requiresContainer = true;
+        $this->requiresStateInputs = true;
+    }
+
+    public function getDeploymentPackage()
+    {
+        $package = parent::getDeploymentPackage();
+        $package->resourcesToDeploy[] = __DIR__ . "/AddressUkPafLookupViewBridge.js";
+
+        return $package;
+    }
+
+    protected function getClientSideViewBridgeName()
+    {
+        return "AddressUkPafLookupViewBridge";
+    }
+
+    public function createPresenters()
+    {
+        $this->AddPresenters(
+            $country = new DropDown("Country"),
+            new TextBox("HouseNumber", 20),
+            new TextBox("PostCodeSearch", 10),
+            $search = new Button("Search", "Search", function() {}),
+            new TextBox("Line1", 50),
+            new TextBox("Line2", 30),
+            new TextBox("Town", 30),
+            new TextBox("County", 20),
+            new TextBox("PostCode", 10)
+        );
+
+        $country->SetSelectionItems( [ [ "", "Please select..." ], Country::getCountriesList() ] );
+    }
+
+    public function printViewContent()
+    {
+        $this->printFieldset( "", [ "Country" ] );
```

Comment:
On 5 Oct 2015, at 19:33, Niall <notifications@github.com> wrote:

> In src/AddressUkPafLookupView.php:
> 
> > +            new TextBox("HouseNumber", 20),
> > +            new TextBox("PostCodeSearch", 10),
> > +            $search = new Button("Search", "Search", function() {}),
> > +            new TextBox("Line1", 50),
> > +            new TextBox("Line2", 30),
> > +            new TextBox("Town", 30),
> > +            new TextBox("County", 20),
> > +            new TextBox("PostCode", 10)
> > +        );
> > +
> > +        $country->SetSelectionItems( [ [ "", "Please select..." ], Country::getCountriesList() ] );
> > +    }
> > +
> > +    public function printViewContent()
> > +    {
> > +        $this->printFieldset( "", [ "Country" ] );
> This country input is provided, but not tied into the form at all. Shouldn't it switch to manual address entry if the country is switched away from UK?
> 
> —
> Reply to this email directly or view it on GitHub.
> 


About the country field the point is that the other fields are inside a specific div for JS purpose



Filippo Polanzan Developer
www.gcdtech.com
fpolanzan@gcdtech.com
Twitter | Facebook
028 3834 1205

Head Office  259 Lough Road, Lurgan, Co. Armagh, BT66 6NQ
Belfast Office  25 Talbot Street, Cathedral Quarter, Belfast, BT1 2LD